server/pages/index.js

```diff
@@ -388,14 +400,33 @@ function setUpLoggedInRoute( req, res, next ) {
 					}
 
 					console.error( 'API Error: ' + errorMessage );
-
-					next( error );
 				}
 
-				return;
+				return error;
 			} );
+
+		Promise.all( [ langPromise, userPromise ] ).then( results => {
+			const rejectedWithErrors = results.filter( r => null != r && r.error );
+
+			if ( 0 !== rejectedWithErrors.length ) {
+				return next( rejectedWithErrors );
+			}
+
+			next();
+		} );
 	} else {
-		next();
+		fetchLangRevisions()
```

Comment:
Can we avoid the duplication of `fetchLangRevisions()` here? Maybe by creating a promises array to contain `langPromise` before the `if ( config.isEnabled( 'wpcom-user-bootstrap' ) ) {` conditional, and inside that conditional, appending `userPromise` to the array. Then finally, we'd need to move the `Promises.all()` thing below that conditional.